python/amici/petab_objective.py

```diff
@@ -1,181 +1,588 @@
+        parameter_mapping: List[petab.ParMappingDictTuple] = None,
+        parameter_scale_mapping: List[petab.ScaleMappingDictTuple] = None,
+        scaled_parameters: Optional[bool] = False,
+        log_level: int = logging.WARNING
+) -> Dict[str, Any]:
+    """Simulate PEtab model
+
+    Arguments:
+        petab_problem:
+            PEtab problem to work on.
+        amici_model:
+            AMICI Model assumed to be compatible with ``petab_problem``.
+        solver:
+            An AMICI solver. Will use default options if None.
+        problem_parameters:
+            Run simulation with these parameters. If None,
+            PEtab `nominalValues` will be used). To be provided as dict,
+            mapping PEtab problem parameters to SBML IDs.
+        simulation_conditions:
+            Result of petab.get_simulation_conditions. Can be provided to save
+            time if this has be obtained before.
+        parameter_mapping:
+            Optional precomputed PEtab parameter mapping for efficiency.
+        parameter_scale_mapping:
+            Optional precomputed PEtab parameter scale mapping for efficiency.
+        scaled_parameters:
+            If True, problem_parameters are assumed to be on the scale provided
+            in the PEtab parameter table and will be unscaled. If False, they
+            are assumed to be in linear scale.
+        log_level:
+            Log level, see `logging` module.
+    Returns:
+        Tuple of cost function value and a list of `ReturnData`s, corresponding
+        to the different simulation conditions. For ordering of simulation
+        conditions, see
+        `petab.Problem.get_simulation_conditions_from_measurement_df`.
+    """
+    logger.setLevel(log_level)
+
+    if solver is None:
+        solver = amici_model.getSolver()
+
+    # Get parameters
+    if problem_parameters is None:
+        # Use PEtab nominal values as default
+        problem_parameters = {t.Index: getattr(t, NOMINAL_VALUE) for t in
+                              petab_problem.parameter_df.itertuples()}
+
+    # Get parameter mapping
+    if parameter_mapping is None:
+        parameter_mapping = \
+            petab_problem.get_optimization_to_simulation_parameter_mapping(
+                warn_unmapped=False)
+
+    # Generate ExpData with all condition-specific information
+    edatas = edatas_from_petab(
+        model=amici_model,
+        petab_problem=petab_problem,
+        problem_parameters=problem_parameters,
+        simulation_conditions=simulation_conditions,
+        parameter_mapping=parameter_mapping,
+        parameter_scale_mapping=parameter_scale_mapping,
+        scaled_parameters=scaled_parameters)
+
+    # Simulate
+    rdatas = amici.runAmiciSimulations(amici_model, solver, edata_list=edatas)
+
+    # Compute total llh
+    llh = sum(rdata['llh'] for rdata in rdatas)
+    # Compute total sllh
+    sllh = aggregate_sllh(amici_model=amici_model, rdatas=rdatas,
```

Comment:
is there a need to set the solver sensitivity order s.t. sllhs are returned?

Reply:
It will be NaN, if you did not enable sensitivities on `solver`.

Reply:
one could (_future_) also allow passing e.g. ``mode=[MODE_RES|MODE_FUN]`` and ``sensi_orders=(0,2,3)`` or ``max_sensi_order=2``, similar to how it's done in pypesto.